webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/DataSourceController.scala

```diff
@@ -484,7 +484,7 @@ Expects:
           organizationName)
         for {
           _ <- Fox.runIf(layerName.isDefined)(
-            dataSourceRepository.invalidateVaultCache(organizationName, dataSetName, layerName.getOrElse("")))
+            dataSourceService.invalidateVaultCache(reloadedDataSource, layerName.getOrElse("")))
```

Comment:
```suggestion
            dataSourceService.invalidateVaultCache(reloadedDataSource, layerName))
```
Would it be possible to do all layers if no layer name is passed to the route?

Reply:
> Thanks for having a look at this! I think the structure is good already :) Please see my two comments. Also: could this return the number of removed keys? Then we could add that to the logging in the controller.

For the number of keys? Do we need to check if it exists before removing or would it be fine to assume it exists, so counting every remove call as one removed vault. In that path, the number will just be the number of mags

Reply:
I guess the number of mags will do for the moment. I added a note to https://github.com/scalableminds/webknossos/issues/6969 that the cache API’s remove should return the number of removed items (0 or 1). If we add that there later we can make this more exact here